### ISSUE TITLE
Fail if au_GetLatest returns an array

### ISSUE THREAD

**Issue by ComFreek** [majkinetor/au#18]
It is very easy to forget discarding the return type of some commands inside GetLatest which make it return an array. This in turn results in failure inside AU:
```
example-package - checking updates
Can't validate latest example-package URL (disable using $NoCheckUrl) ''
The URL is empty
In C:\Program Files\WindowsPowerShell\Modules\au\2016.8.15\Update-Package.ps1:108 Zeichen:26
+ ... f (!$res) { throw "Can't validate latest $packageName URL (disable us ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Can't validate ...he URL is empty:String) [], RuntimeException
    + FullyQualifiedErrorId : Can't validate latest netbeans URL (disable using $NoCheckUrl) ''
The URL is empty
```

Personally, I encountered this and was puzzled about it when I had multiple `-match` statements:
```powershell
$hashPage = Invoke-WebRequest ...
$hashPage.Content -match '...' # Missing | Out-Null!
```

I would suggest
- making AU fail when GetLatest returns an array
- and/or update the documentation.

What do you think?

**Comment by majkinetor** [comment]
Well, AU failed didn't it ? With the message that URL is missing.

I think its clear enough.

You can always write $Latest on output to examine it inside `au_GetLatest` (it would probably be good to show it in full with Verbose).

**Comment by majkinetor** [comment]
Also, in your case it still returned hashtable, just the URL key was an array.

**Comment by ComFreek** [comment]
> Well, AU failed didn't it ? With the message that URL is missing. I think its clear enough.

The error message was more puzzling than helpful for me since `Write-Host` - as a debugging measure - verified the apparently correct return value

> You can always write $Latest on output to examine it inside  au_GetLatest 

I did:

```powershell
function global:au_GetLatest {
  $hashPage = Invoke-WebRequest ...
  $hashPage.Content -match '...'

  $version = $matches[1]
  $finalUri = "...$version..."
  Write-Host $finalUri # non-empty value
  @{
    Version = $version
    URL32 = $finalUri # now it's an empty value, according to the error message?!
  }
}
```

> Also, in your case it still returned hashtable, just the URL key was an array. 

```powershell
function abc {
  'a' -match 'a'
  @{
    URL32 = 'google.de'
    Version = '1.0'
  }
}
```
`(abc).GetType()` returns `Object[]`, `(abc)[0]` returns `True` and `(abc)[1]` returns the actual hash table.

**Comment by majkinetor** [comment]
> Write-Host - as a debugging measure - verified the apparently correct return value

Not different then usual posh troubleshooting. 

> function abc

That is entirely different example, the function returns 2 values and behavior is normal. That is not what happened with your example and checking for return type wouldn't make a difference. 

I will add the HashTable return value check as it does seem protective.

**Comment by majkinetor** [comment]
I made various improvements and `$Latest` is now shown along with the key type and value.

```
15  16-08-30 11:20:04  C:\Work\chocolatey\dngrep
> $au_force = $true; $au_nocheckurl = $true; import-module au -force; ./update.ps1
dngrep - checking updates
nuspec version: 2.8.16.20160830
remote version: 2.8.16.0
No new version found, but update is forced
Automatic checksum started
Downloading dngrep 32 bit
  from 'https://github.com/dnGrep/dnGrep/releases/download/v2.8.16.0/dnGREP.2.8.16.x86.msi'

Download of dnGREP.2.8.16.x86.msi (3,36 MB) completed.
Package downloaded and hash calculated for 32 bit version
Downloading dngrep 64 bit
  from 'https://github.com/dnGrep/dnGrep/releases/download/v2.8.16.0/dnGREP.2.8.16.x64.msi'

Download of dnGREP.2.8.16.x64.msi (3,39 MB) completed.
Package downloaded and hash calculated for 64 bit version
Updating files
  $Latest data:
    Checksum32      (String)    ce4753735148e1f48fe0e1cd9aa4dfd019082f4f43c38c4ff4157f08d346700c
    Checksum64      (String)    025bd4101826932e954aacd3fe6aee9927a7198feeffb24f82fbe5d578502d18
    ChecksumType32  (String)    sha256
    ChecksumType64  (String)    sha256
    NuspecVersion   (String)    2.8.16.20160830
    PackageName     (String)    dngrep
    URL32           (String)    https://github.com/dnGrep/dnGrep/releases/download/v2.8.16.0/dnGREP.2.8.16.x86.msi
    URL64           (String)    https://github.com/dnGrep/dnGrep/releases/download/v2.8.16.0/dnGREP.2.8.16.x64.msi
    Version         (String)    2.8.16.0

  dngrep.nuspec
    updating version using Chocolatey fix notation: 2.8.16.20160830 -> 2.8.16.20160830
  tools\chocolateyInstall.ps1
    (^[$]url64\s*=\s*)('.*') = $1'https://github.com/dnGrep/dnGrep/releases/download/v2.8.16.0/dnGREP.2.8.16.x64.msi'
    (^[$]url32\s*=\s*)('.*') = $1'https://github.com/dnGrep/dnGrep/releases/download/v2.8.16.0/dnGREP.2.8.16.x86.msi'
    (^[$]checksum32\s*=\s*)('.*') = $1'ce4753735148e1f48fe0e1cd9aa4dfd019082f4f43c38c4ff4157f08d346700c'
    (^[$]checksum64\s*=\s*)('.*') = $1'025bd4101826932e954aacd3fe6aee9927a7198feeffb24f82fbe5d578502d18'
Chocolatey v0.10.0
Attempting to build package from 'dngrep.nuspec'.
Successfully created package 'C:\Work\chocolatey\dngrep\dngrep.2.8.16.20160830.nupkg'
Package updated
```
